### PR TITLE
Feature/story 22

### PR DESCRIPTION
**Pull request by Edfreded**
Added

- Register endpoint
- Login endpoint
- Authorized only endpoint

First create an account (http://localhost:8080/index.html)
Untitled

Then login
login

add the bearer token to swagger and fetch the test endpoint
token

### CODE REVIEW

File: Assemblies/AllYourGoods.Api/Data/ApplicationContext.cs [hunk 1]
```diff
@@ -15,12 +17,16 @@ public ApplicationContext()
     }
 
     public virtual DbSet<Restaurant> Restaurants { get; set; } = null!;
+    // public override DbSet<User> Users { get; set; } = null!;
```
Comment (KiwiLT) [inline Assemblies/AllYourGoods.Api/Data/ApplicationContext.cs:20]: old comment

File: Assemblies/AllYourGoods.Api/Controllers/RestaurantController.cs [hunk 2]
```diff
@@ -1,6 +1,10 @@
 ﻿using AllYourGoods.Api.Interfaces.Services;
 using AllYourGoods.Api.Models.Dtos;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using Microsoft.AspNetCore.Authorization;
```
Comment (KiwiLT) [inline Assemblies/AllYourGoods.Api/Controllers/RestaurantController.cs:7]: Zijn deze imports nodig in Restaurant controller?

File: Assemblies/AllYourGoods.Api/Controllers/TestController.cs
Comment (KiwiLT) [inline Assemblies/AllYourGoods.Api/Controllers/TestController.cs:1]: Moet weg of moet een test worden

File: Assemblies/AllYourGoods.Api/Data/ApplicationContext.cs [hunk 3]
```diff
@@ -1,9 +1,11 @@
 ﻿using AllYourGoods.Api.Models;
+using Microsoft.AspNetCore.Identity;
```
Comment (KiwiLT) [inline Assemblies/AllYourGoods.Api/Data/ApplicationContext.cs:2]: dubbele import

File: Assemblies/AllYourGoods.Api/Program.cs [hunk 4]
```diff
@@ -70,8 +77,63 @@ private static WebApplicationBuilder CreateBuilder(string[] args)
         builder.Services.AddHealthChecks()
             .AddDbContextCheck<ApplicationContext>();
 
+        // builder.Services.AddIdentity<User, IdentityRole>()
```
Comment (KiwiLT) [inline Assemblies/AllYourGoods.Api/Program.cs:80]: old comment

File: Assemblies/AllYourGoods.Api/Program.cs [hunk 5]
```diff
@@ -70,8 +77,63 @@ private static WebApplicationBuilder CreateBuilder(string[] args)
...
+                ValidateLifetime = true,
+                ValidateIssuerSigningKey = true,
+                ValidIssuer = builder.Configuration["Jwt:Issuer"],
+                ValidAudience = builder.Configuration["Jwt:Audience"],
+                IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(builder.Configuration["Jwt:Key"]))
+            };
+        });
+
+        builder.Services.AddHealthChecks();
+        // Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
```
Comment (KiwiLT) [inline Assemblies/AllYourGoods.Api/Program.cs:105]: weird comment

File: Assemblies/AllYourGoods.Api/Program.cs [hunk 6]
```diff
@@ -70,8 +77,63 @@ private static WebApplicationBuilder CreateBuilder(string[] args)
...
         builder.Services.AddEndpointsApiExplorer();
-        builder.Services.AddSwaggerGen();
+        // builder.Services.AddSwaggerGen();
```
Comment (KiwiLT) [inline Assemblies/AllYourGoods.Api/Program.cs:107]: old comment

**Review by KiwiLT — CHANGES_REQUESTED** [review]
few minor code cleanups, everything worked when testing

**Comment by Edfreded** [comment]
critical changes made. pr obsolete

**Pull request closed without merging**